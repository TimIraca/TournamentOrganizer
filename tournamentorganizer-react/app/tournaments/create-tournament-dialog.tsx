"use client";

import { useState } from "react";
import { Button } from "@/components/ui/button";
import { Input } from "@/components/ui/input";
import { Label } from "@/components/ui/label";
import {
  Dialog,
  DialogContent,
  DialogDescription,
  DialogFooter,
  DialogHeader,
  DialogTitle,
  DialogTrigger,
} from "@/components/ui/dialog";
import { toast } from "@/hooks/use-toast";

import { editTournament } from "@/types";
import { PlusCircle } from "lucide-react";

interface CreateTournamentDialogProps {
  onCreateTournament: (tournament: editTournament) => void;
}

export function CreateTournamentDialog({
  onCreateTournament,
}: CreateTournamentDialogProps) {
  const [open, setOpen] = useState(false);
  const [name, setName] = useState("");
  const [startDate, setStartDate] = useState<string>("");

  const handleCreateTournament = () => {
    if (!name || !startDate) {
      toast({
        title: "Error",
        description:
          "Please provide both a name and a start date for the tournament.",
        variant: "destructive",
      });
      return;
    }

    const newTournament: editTournament = {
      name,
      startDate: new Date(startDate).toISOString(),
    };

    onCreateTournament(newTournament);
    setOpen(false);
    setName("");
    setStartDate("");
    toast({
      title: "Success",
      description: "New tournament created successfully.",
    });
  };

  return (
    <Dialog open={open} onOpenChange={setOpen}>
      <DialogTrigger asChild>
        <Button data-cy="create-tournament-button">
          <PlusCircle className="mr-2 h-4 w-4" />
          New Tournament
        </Button>
      </DialogTrigger>
      <DialogContent className="sm:max-w-[425px]">
        <DialogHeader>
          <DialogTitle>Create New Tournament</DialogTitle>
          <DialogDescription>
            Enter the details for the new tournament. Click create when
            you&apos;re done.
          </DialogDescription>
        </DialogHeader>
        <div className="grid gap-4 py-4">
          <div className="grid grid-cols-4 items-center gap-4">
            <Label htmlFor="name" className="text-right">
              Name
            </Label>
            <Input
              id="name"
              value={name}
              onChange={(e) => setName(e.target.value)}
              className="col-span-3"
            />
          </div>
          <div className="grid grid-cols-4 items-center gap-4">
            <Label htmlFor="startDate" className="text-right">
              Start Date
            </Label>
            <Input
              type="date"
              id="startDate"
              value={startDate}
              onChange={(e) => setStartDate(e.target.value)}
              className="col-span-3"
            />
          </div>
        </div>
        <DialogFooter>
          <Button type="submit" onClick={handleCreateTournament}>
            Create Tournament
          </Button>
        </DialogFooter>
      </DialogContent>
    </Dialog>
  );
}
