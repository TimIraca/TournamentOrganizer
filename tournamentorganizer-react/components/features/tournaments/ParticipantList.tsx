import {
  Table,
  TableBody,
  TableCell,
  TableHead,
  TableHeader,
  TableRow,
} from "@/components/ui/table";
import { Button } from "@/components/ui/button";
import { Participant } from "@/types";
import { formatDate } from "@/lib/utils";

interface ParticipantListProps {
  participants: Participant[];
  maxParticipants: number;
  onRemoveParticipant?: (participantId: string) => Promise<void>;
}

export function ParticipantList({
  participants,
  maxParticipants,
  onRemoveParticipant,
}: ParticipantListProps) {
  return (
    <div className="space-y-4">
      <div className="flex items-center justify-between">
        <h3 className="text-lg font-medium">
          Participants ({participants.length}/{maxParticipants})
        </h3>
      </div>
      <Table>
        <TableHeader>
          <TableRow>
            <TableHead>Name</TableHead>
            <TableHead>Registration Date</TableHead>
            <TableHead></TableHead>
          </TableRow>
        </TableHeader>
        <TableBody>
          {participants.map((participant) => (
            <TableRow key={participant.id}>
              <TableCell>{participant.participantName}</TableCell>
              <TableCell>{formatDate(participant.registrationDate)}</TableCell>
              <TableCell className="text-right">
                {onRemoveParticipant && (
                  <Button
                    variant="ghost"
                    size="sm"
                    onClick={() => onRemoveParticipant(participant.id)}
                  >
                    Remove
                  </Button>
                )}
              </TableCell>
            </TableRow>
          ))}
        </TableBody>
      </Table>
    </div>
  );
}
