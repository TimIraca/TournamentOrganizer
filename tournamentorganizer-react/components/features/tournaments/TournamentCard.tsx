import {
  Card,
  CardContent,
  CardHeader,
  CardTitle,
  CardDescription,
} from "@/components/ui/card";
import { Badge } from "@/components/ui/badge";
import { Tournament } from "@/types";
import { formatDate } from "@/lib/utils"; // We'll create this utility

interface TournamentCardProps {
  tournament: Tournament;
  onClick: (tournament: Tournament) => void;
}

export function TournamentCard({ tournament, onClick }: TournamentCardProps) {
  return (
    <Card
      className="cursor-pointer hover:bg-accent/50"
      onClick={() => onClick(tournament)}
    >
      <CardHeader>
        <CardTitle>{tournament.name}</CardTitle>
        <CardDescription>{formatDate(tournament.startDate)}</CardDescription>
      </CardHeader>
      <CardContent>
        <div className="flex flex-col gap-2">
          <div className="flex items-center justify-between">
            <span className="text-sm text-muted-foreground">
              {tournament.currentParticipants}/{tournament.maxParticipants}{" "}
              Participants
            </span>
            <Badge
              variant={
                tournament.status === "Registration" ? "default" : "secondary"
              }
            >
              {tournament.status}
            </Badge>
          </div>
          <div className="flex items-center justify-between">
            <span className="text-sm text-muted-foreground">
              Prize Pool: {tournament.prizePool} {tournament.prizeCurrency}
            </span>
            <span className="text-sm font-medium">{tournament.format}</span>
          </div>
        </div>
      </CardContent>
    </Card>
  );
}
