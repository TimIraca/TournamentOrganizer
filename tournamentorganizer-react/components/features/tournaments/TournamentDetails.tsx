// components/features/tournaments/TournamentDetails.tsx
import { useState } from "react";
import {
  Card,
  CardHeader,
  CardTitle,
  CardDescription,
  CardContent,
} from "@/components/ui/card";
import { Input } from "@/components/ui/input";
import { Button } from "@/components/ui/button";
import { Badge } from "@/components/ui/badge";
import { handleApiError } from "@/lib/api";
import type { Tournament, TournamentStatus } from "@/types";
import { formatDate } from "@/lib/utils";

interface TournamentDetailsProps {
  tournament: Tournament;
  onStartTournament?: () => Promise<void>;
}

export function TournamentDetails({
  tournament,
  onStartTournament,
}: TournamentDetailsProps) {
  const [isEditing, setIsEditing] = useState(false);
  const [formData, setFormData] = useState({
    name: tournament.name,
    description: tournament.description,
    format: tournament.format,
    startDate: tournament.startDate,
    maxParticipants: tournament.maxParticipants,
    prizePool: tournament.prizePool,
    prizeCurrency: tournament.prizeCurrency,
  });

  const handleSubmit = async (e: React.FormEvent) => {
    e.preventDefault();
    try {
      // Assuming you'll add an updateTournament endpoint to your API client
      // await tournamentApi.updateTournament(tournament.id, formData);
      setIsEditing(false);
    } catch (error) {
      console.error("Failed to update tournament:", handleApiError(error));
    }
  };

  const formatStatusBadge = (
    status: TournamentStatus
  ): "default" | "secondary" | "outline" | "destructive" | null | undefined => {
    const variants: Record<
      TournamentStatus,
      "default" | "secondary" | "outline"
    > = {
      Registration: "default",
      InProgress: "secondary",
      Completed: "outline",
    };
    return variants[status];
  };

  if (!isEditing) {
    return (
      <Card>
        <CardHeader className="flex flex-row items-center justify-between space-y-0 pb-2">
          <div className="space-y-1">
            <CardTitle>{tournament.name}</CardTitle>
            <CardDescription>
              {formatDate(tournament.startDate)}
            </CardDescription>
          </div>
          <Badge variant={formatStatusBadge(tournament.status)}>
            {tournament.status}
          </Badge>
        </CardHeader>
        <CardContent>
          <div className="space-y-4">
            <div className="grid gap-1">
              <label className="text-sm font-medium">Description</label>
              <p className="text-sm text-muted-foreground">
                {tournament.description}
              </p>
            </div>

            <div className="grid grid-cols-2 gap-4">
              <div className="grid gap-1">
                <label className="text-sm font-medium">Format</label>
                <p className="text-sm text-muted-foreground">
                  {tournament.format}
                </p>
              </div>
              <div className="grid gap-1">
                <label className="text-sm font-medium">Participants</label>
                <p className="text-sm text-muted-foreground">
                  {tournament.currentParticipants} /{" "}
                  {tournament.maxParticipants}
                </p>
              </div>
            </div>

            <div className="grid grid-cols-2 gap-4">
              <div className="grid gap-1">
                <label className="text-sm font-medium">Prize Pool</label>
                <p className="text-sm text-muted-foreground">
                  {tournament.prizePool} {tournament.prizeCurrency}
                </p>
              </div>
            </div>

            <div className="flex gap-2">
              {tournament.status === "Registration" && onStartTournament && (
                <Button onClick={onStartTournament} className="flex-1">
                  Start Tournament
                </Button>
              )}
              <Button
                variant="outline"
                onClick={() => setIsEditing(true)}
                className="flex-1"
              >
                Edit Tournament
              </Button>
            </div>
          </div>
        </CardContent>
      </Card>
    );
  }

  return (
    <Card>
      <CardHeader>
        <CardTitle>Edit Tournament</CardTitle>
        <CardDescription>Update tournament details</CardDescription>
      </CardHeader>
      <CardContent>
        <form onSubmit={handleSubmit} className="space-y-4">
          <div className="grid gap-2">
            <label className="text-sm font-medium">Tournament Name</label>
            <Input
              value={formData.name}
              onChange={(e) =>
                setFormData((prev) => ({ ...prev, name: e.target.value }))
              }
            />
          </div>

          <div className="grid gap-2">
            <label className="text-sm font-medium">Description</label>
            <Input
              value={formData.description}
              onChange={(e) =>
                setFormData((prev) => ({
                  ...prev,
                  description: e.target.value,
                }))
              }
            />
          </div>

          {/* <div className="grid gap-2">
            <label className="text-sm font-medium">Format</label>
            <Select
              value={formData.format}
              onValueChange={(value: TournamentFormat) => 
                setFormData(prev => ({ ...prev, format: value }))
              }
            >
              <SelectTrigger>
                <SelectValue placeholder="Select format" />
              </SelectTrigger>
              <SelectContent>
                <SelectItem value="SingleElimination">Single Elimination</SelectItem>
                <SelectItem value="RoundRobin">Round Robin</SelectItem>
              </SelectContent>
            </Select>
          </div> */}

          <div className="grid gap-2">
            <label className="text-sm font-medium">Start Date</label>
            <Input
              type="datetime-local"
              value={formData.startDate.split(".")[0]} // Remove milliseconds if present
              onChange={(e) =>
                setFormData((prev) => ({ ...prev, startDate: e.target.value }))
              }
            />
          </div>

          <div className="grid gap-2">
            <label className="text-sm font-medium">Maximum Participants</label>
            <Input
              type="number"
              value={formData.maxParticipants}
              onChange={(e) =>
                setFormData((prev) => ({
                  ...prev,
                  maxParticipants: parseInt(e.target.value, 10),
                }))
              }
            />
          </div>

          <div className="grid grid-cols-2 gap-4">
            <div className="grid gap-2">
              <label className="text-sm font-medium">Prize Pool</label>
              <Input
                type="number"
                value={formData.prizePool}
                onChange={(e) =>
                  setFormData((prev) => ({
                    ...prev,
                    prizePool: parseInt(e.target.value, 10),
                  }))
                }
              />
            </div>
            <div className="grid gap-2">
              <label className="text-sm font-medium">Currency</label>
              <Input
                value={formData.prizeCurrency}
                onChange={(e) =>
                  setFormData((prev) => ({
                    ...prev,
                    prizeCurrency: e.target.value,
                  }))
                }
              />
            </div>
          </div>

          <div className="flex gap-2">
            <Button type="submit" className="flex-1">
              Save Changes
            </Button>
            <Button
              type="button"
              variant="outline"
              onClick={() => setIsEditing(false)}
              className="flex-1"
            >
              Cancel
            </Button>
          </div>
        </form>
      </CardContent>
    </Card>
  );
}
