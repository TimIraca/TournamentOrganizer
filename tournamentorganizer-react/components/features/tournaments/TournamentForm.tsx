import React from "react";
import { Card, CardHeader, CardTitle, CardContent } from "@/components/ui/card";
import { Input } from "@/components/ui/input";
import { Button } from "@/components/ui/button";
import { Textarea } from "@/components/ui/textarea";
import {
  Select,
  SelectContent,
  SelectItem,
  SelectTrigger,
  SelectValue,
} from "@/components/ui/select";
import {
  Form,
  FormControl,
  FormDescription,
  FormField,
  FormItem,
  FormLabel,
  FormMessage,
} from "@/components/ui/form";
import { Switch } from "@/components/ui/switch";
import { useForm } from "react-hook-form";
import { zodResolver } from "@hookform/resolvers/zod";
import * as z from "zod";
import { CreateTournamentRequest, TournamentFormat } from "@/types";

const tournamentFormSchema = z.object({
  name: z
    .string()
    .min(3, "Tournament name must be at least 3 characters")
    .max(100, "Tournament name must be less than 100 characters"),
  description: z.string().optional(),
  format: z.nativeEnum(TournamentFormat),
  startDate: z.string().refine((date) => new Date(date) > new Date(), {
    message: "Start date must be in the future",
  }),
  maxParticipants: z
    .number()
    .min(2, "Tournament must have at least 2 participants")
    .max(128, "Tournament cannot have more than 128 participants"),
  hasPrizePool: z.boolean().default(false),
  prizePool: z.number().min(0, "Prize pool cannot be negative").optional(),
  prizeCurrency: z.string().optional(),
});

type TournamentFormData = z.infer<typeof tournamentFormSchema>;

interface TournamentFormProps {
  onSubmit: (data: CreateTournamentRequest) => Promise<void>;
  isLoading?: boolean;
}

export function TournamentForm({ onSubmit, isLoading }: TournamentFormProps) {
  // Get tomorrow's date at current time
  const tomorrow = new Date();
  tomorrow.setDate(tomorrow.getDate() + 1);
  const tomorrowString = tomorrow.toISOString().slice(0, 16); // Format: YYYY-MM-DDTHH:mm

  const form = useForm<TournamentFormData>({
    resolver: zodResolver(tournamentFormSchema),
    defaultValues: {
      name: "",
      description: "",
      format: TournamentFormat.SingleElimination,
      startDate: tomorrowString,
      maxParticipants: 16,
      hasPrizePool: false,
      prizePool: 0,
      prizeCurrency: "USD",
    },
  });

  const hasPrizePool = form.watch("hasPrizePool");

  const handleSubmit = async (data: TournamentFormData) => {
    const submitData: CreateTournamentRequest = {
      name: data.name,
      description: data.description,
      format: data.format,
      startDate: new Date(data.startDate).toISOString(),
      maxParticipants: data.maxParticipants,
      prizePool: data.hasPrizePool ? data.prizePool : 0,
      prizeCurrency: data.hasPrizePool ? data.prizeCurrency : "USD",
    };

    await onSubmit(submitData);
  };

  return (
    <Card className="w-full max-w-2xl mx-auto">
      <CardHeader>
        <CardTitle>Create New Tournament</CardTitle>
      </CardHeader>
      <CardContent>
        <Form {...form}>
          <form
            onSubmit={form.handleSubmit(handleSubmit)}
            className="space-y-6"
          >
            <FormField
              control={form.control}
              name="name"
              render={({ field }) => (
                <FormItem>
                  <FormLabel>Tournament Name</FormLabel>
                  <FormControl>
                    <Input placeholder="Enter tournament name" {...field} />
                  </FormControl>
                  <FormDescription>
                    Choose a unique and descriptive name
                  </FormDescription>
                  <FormMessage />
                </FormItem>
              )}
            />

            <FormField
              control={form.control}
              name="description"
              render={({ field }) => (
                <FormItem>
                  <FormLabel>Description (Optional)</FormLabel>
                  <FormControl>
                    <Textarea
                      placeholder="Enter tournament description"
                      className="resize-none"
                      {...field}
                    />
                  </FormControl>
                  <FormMessage />
                </FormItem>
              )}
            />

            <FormField
              control={form.control}
              name="format"
              render={({ field }) => (
                <FormItem>
                  <FormLabel>Tournament Format</FormLabel>
                  <Select
                    onValueChange={(value) => field.onChange(Number(value))}
                    defaultValue={field.value.toString()}
                  >
                    <FormControl>
                      <SelectTrigger>
                        <SelectValue placeholder="Select format" />
                      </SelectTrigger>
                    </FormControl>
                    <SelectContent>
                      <SelectItem
                        value={TournamentFormat.SingleElimination.toString()}
                      >
                        Single Elimination
                      </SelectItem>
                      <SelectItem
                        value={TournamentFormat.DoubleElimination.toString()}
                      >
                        Double Elimination
                      </SelectItem>
                      <SelectItem
                        value={TournamentFormat.RoundRobin.toString()}
                      >
                        Round Robin
                      </SelectItem>
                    </SelectContent>
                  </Select>
                  <FormDescription>
                    Choose the tournament bracket format
                  </FormDescription>
                  <FormMessage />
                </FormItem>
              )}
            />

            <FormField
              control={form.control}
              name="startDate"
              render={({ field }) => (
                <FormItem>
                  <FormLabel>Start Date</FormLabel>
                  <FormControl>
                    <Input
                      type="datetime-local"
                      {...field}
                      min={new Date().toISOString().slice(0, 16)}
                    />
                  </FormControl>
                  <FormDescription>
                    When will the tournament begin?
                  </FormDescription>
                  <FormMessage />
                </FormItem>
              )}
            />

            <FormField
              control={form.control}
              name="maxParticipants"
              render={({ field }) => (
                <FormItem>
                  <FormLabel>Maximum Participants</FormLabel>
                  <FormControl>
                    <Input
                      type="number"
                      {...field}
                      onChange={(e) => field.onChange(parseInt(e.target.value))}
                    />
                  </FormControl>
                  <FormDescription>
                    Number of participants (2-128)
                  </FormDescription>
                  <FormMessage />
                </FormItem>
              )}
            />

            <FormField
              control={form.control}
              name="hasPrizePool"
              render={({ field }) => (
                <FormItem className="flex flex-row items-center justify-between rounded-lg border p-4">
                  <div className="space-y-0.5">
                    <FormLabel className="text-base">Prize Pool</FormLabel>
                    <FormDescription>
                      Enable if this tournament has prizes
                    </FormDescription>
                  </div>
                  <FormControl>
                    <Switch
                      checked={field.value}
                      onCheckedChange={field.onChange}
                    />
                  </FormControl>
                </FormItem>
              )}
            />

            {hasPrizePool && (
              <div className="space-y-4">
                <FormField
                  control={form.control}
                  name="prizePool"
                  render={({ field }) => (
                    <FormItem>
                      <FormLabel>Prize Amount</FormLabel>
                      <FormControl>
                        <Input
                          type="number"
                          {...field}
                          onChange={(e) =>
                            field.onChange(parseFloat(e.target.value))
                          }
                          min={0}
                          step="0.01"
                        />
                      </FormControl>
                      <FormMessage />
                    </FormItem>
                  )}
                />

                <FormField
                  control={form.control}
                  name="prizeCurrency"
                  render={({ field }) => (
                    <FormItem>
                      <FormLabel>Currency</FormLabel>
                      <Select
                        onValueChange={field.onChange}
                        defaultValue={field.value}
                      >
                        <FormControl>
                          <SelectTrigger>
                            <SelectValue placeholder="Select currency" />
                          </SelectTrigger>
                        </FormControl>
                        <SelectContent>
                          <SelectItem value="USD">USD</SelectItem>
                          <SelectItem value="EUR">EUR</SelectItem>
                          <SelectItem value="GBP">GBP</SelectItem>
                        </SelectContent>
                      </Select>
                      <FormMessage />
                    </FormItem>
                  )}
                />
              </div>
            )}

            <Button type="submit" className="w-full" disabled={isLoading}>
              {isLoading ? "Creating..." : "Create Tournament"}
            </Button>
          </form>
        </Form>
      </CardContent>
    </Card>
  );
}

export default TournamentForm;
