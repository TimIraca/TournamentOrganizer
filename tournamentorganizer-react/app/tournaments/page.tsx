"use client";
import { useState, useEffect } from "react";
import { TournamentCard } from "@/components/Tournaments/TournamentCard";
import { editTournament, Tournament } from "@/types";
import { Edit, Trash } from "lucide-react";
import { Skeleton } from "@/components/ui/skeleton";
import { Alert, AlertDescription } from "@/components/ui/alert";
import { SidebarTrigger } from "@/components/ui/sidebar";
import { tournamentApi } from "@/lib/api";
import { ModeToggle } from "@/components/modetoggle";
import Link from "next/link";
import { GeneralContextMenu } from "@/components/general-context-menu";
import {
  AlertDialog,
  AlertDialogAction,
  AlertDialogCancel,
  AlertDialogContent,
  AlertDialogDescription,
  AlertDialogFooter,
  AlertDialogHeader,
  AlertDialogTitle,
} from "@/components/ui/alert-dialog";
import { CreateTournamentDialog } from "./create-tournament-dialog";

export default function Page() {
  const [tournaments, setTournaments] = useState<Tournament[]>([]);
  const [loading, setLoading] = useState(true);
  const [error, setError] = useState<string | null>(null);
  const [tournamentToDelete, setTournamentToDelete] = useState<string | null>(
    null
  );
  const handleDeleteConfirm = async () => {
    if (tournamentToDelete) {
      try {
        await tournamentApi.deleteTournament(tournamentToDelete);
        setTournaments(tournaments.filter((t) => t.id !== tournamentToDelete));
        setTournamentToDelete(null);
      } catch (err) {
        setError(`Failed to delete tournament: ${err}`);
      }
    }
  };
  const handleCreateTournament = async (newTournament: editTournament) => {
    try {
      await tournamentApi.createTournament(newTournament);
    } catch (err) {
      setError(`Failed to create tournament: ${err}`);
    } finally {
      fetchTournaments();
    }
  };
  const fetchTournaments = async () => {
    try {
      const data = await tournamentApi.getAllTournaments();
      setTournaments(data);
    } catch {
      setError("Failed to create tournament:");
    } finally {
      setLoading(false);
    }
  };
  useEffect(() => {
    fetchTournaments();
  }, []);
  return (
    <div className="flex flex-col gap-6 p-6 w-full">
      <div className="flex items-center justify-between">
        <div className="flex items-center space-x-4">
          <SidebarTrigger />
          <ModeToggle />
        </div>
        <h1 className="text-3xl font-bold tracking-tight text-center flex-grow">
          Your Tournaments
        </h1>

        <CreateTournamentDialog onCreateTournament={handleCreateTournament} />
      </div>

      {error && (
        <Alert variant="destructive">
          <AlertDescription>{error}</AlertDescription>
        </Alert>
      )}

      {loading ? (
        <div className="grid gap-6 md:grid-cols-2 lg:grid-cols-3">
          {[1, 2, 3].map((n) => (
            <div key={n} className="space-y-3">
              <Skeleton className="h-[125px] w-full rounded-xl" />
            </div>
          ))}
        </div>
      ) : (
        <div className="grid gap-6 md:grid-cols-2 lg:grid-cols-3">
          {tournaments.map((tournament) => (
            <GeneralContextMenu
              key={tournament.id}
              data-cy={tournament.id}
              actions={[
                {
                  label: "Edit",
                  icon: <Edit className="h-4 w-4" />,
                  onClick: (e) => {
                    e.preventDefault();
                    e.stopPropagation();
                  },
                  href: `/tournaments/${tournament.id}/edit`,
                },
                {
                  label: "Delete",
                  icon: <Trash className="h-4 w-4" />,
                  onClick: (e) => {
                    e.preventDefault();
                    e.stopPropagation();
                    setTournamentToDelete(tournament.id);
                  },
                },
              ]}
            >
              <Link
                href={`/tournaments/${tournament.id}`}
                className="block h-full"
              >
                <TournamentCard
                  tournament={tournament}
                  data-cy={tournament.id}
                />
              </Link>
            </GeneralContextMenu>
          ))}
        </div>
      )}
      <AlertDialog
        open={!!tournamentToDelete}
        onOpenChange={() => setTournamentToDelete(null)}
      >
        <AlertDialogContent>
          <AlertDialogHeader>
            <AlertDialogTitle>Are you absolutely sure?</AlertDialogTitle>
            <AlertDialogDescription>
              This action cannot be undone. This will permanently delete the
              tournament and remove all associated data from our servers.
            </AlertDialogDescription>
          </AlertDialogHeader>
          <AlertDialogFooter>
            <AlertDialogCancel>Cancel</AlertDialogCancel>
            <AlertDialogAction onClick={handleDeleteConfirm}>
              Delete
            </AlertDialogAction>
          </AlertDialogFooter>
        </AlertDialogContent>
      </AlertDialog>
    </div>
  );
}
