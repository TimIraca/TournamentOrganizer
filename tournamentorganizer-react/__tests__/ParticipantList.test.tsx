import React from "react";
import { render } from "@testing-library/react";
import { ParticipantList } from "@/components/features/tournaments/ParticipantList";
import "@testing-library/jest-dom";

const mockParticipants = [
  { id: "1", participantName: "Alice", registrationDate: "2023-11-11" },
  { id: "2", participantName: "Bob", registrationDate: "2023-11-12" },
];

describe("ParticipantList", () => {
  it("renders correctly", () => {
    const { container } = render(
      <ParticipantList participants={mockParticipants} maxParticipants={10} />
    );
    expect(container).toMatchSnapshot();
  });
});
