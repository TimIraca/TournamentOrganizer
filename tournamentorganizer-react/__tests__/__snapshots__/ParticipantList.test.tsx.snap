// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ParticipantList renders correctly 1`] = `
<div>
  <div
    class="space-y-4"
  >
    <div
      class="flex items-center justify-between"
    >
      <h3
        class="text-lg font-medium"
      >
        Participants (
        2
        /
        10
        )
      </h3>
    </div>
    <div
      class="relative w-full overflow-auto"
    >
      <table
        class="w-full caption-bottom text-sm"
      >
        <thead
          class="[&_tr]:border-b"
        >
          <tr
            class="border-b transition-colors hover:bg-muted/50 data-[state=selected]:bg-muted"
          >
            <th
              class="h-10 px-2 text-left align-middle font-medium text-muted-foreground [&:has([role=checkbox])]:pr-0 [&>[role=checkbox]]:translate-y-[2px]"
            >
              Name
            </th>
            <th
              class="h-10 px-2 text-left align-middle font-medium text-muted-foreground [&:has([role=checkbox])]:pr-0 [&>[role=checkbox]]:translate-y-[2px]"
            >
              Registration Date
            </th>
            <th
              class="h-10 px-2 text-left align-middle font-medium text-muted-foreground [&:has([role=checkbox])]:pr-0 [&>[role=checkbox]]:translate-y-[2px]"
            />
          </tr>
        </thead>
        <tbody
          class="[&_tr:last-child]:border-0"
        >
          <tr
            class="border-b transition-colors hover:bg-muted/50 data-[state=selected]:bg-muted"
          >
            <td
              class="p-2 align-middle [&:has([role=checkbox])]:pr-0 [&>[role=checkbox]]:translate-y-[2px]"
            >
              Alice
            </td>
            <td
              class="p-2 align-middle [&:has([role=checkbox])]:pr-0 [&>[role=checkbox]]:translate-y-[2px]"
            >
              November 11, 2023 at 01:00 AM
            </td>
            <td
              class="p-2 align-middle [&:has([role=checkbox])]:pr-0 [&>[role=checkbox]]:translate-y-[2px] text-right"
            />
          </tr>
          <tr
            class="border-b transition-colors hover:bg-muted/50 data-[state=selected]:bg-muted"
          >
            <td
              class="p-2 align-middle [&:has([role=checkbox])]:pr-0 [&>[role=checkbox]]:translate-y-[2px]"
            >
              Bob
            </td>
            <td
              class="p-2 align-middle [&:has([role=checkbox])]:pr-0 [&>[role=checkbox]]:translate-y-[2px]"
            >
              November 12, 2023 at 01:00 AM
            </td>
            <td
              class="p-2 align-middle [&:has([role=checkbox])]:pr-0 [&>[role=checkbox]]:translate-y-[2px] text-right"
            />
          </tr>
        </tbody>
      </table>
    </div>
  </div>
</div>
`;
